lib/thredded/collection_to_strings_with_cache_renderer.rb

```diff
@@ -52,12 +61,25 @@ def collection_cache
     end
 
     # @return [Array<String>]
-    def render_partials(view_context, collection:, **opts)
+    def render_partials(view_context, collection:, render_threads:, **opts)
       return [] if collection.empty?
-      partial_renderer = ActionView::PartialRenderer.new(@lookup_context)
-      collection.map do |item|
-        partial_renderer.render(view_context, opts.merge(object: item), nil)
+      num_threads = [render_threads, collection.size].min
+      if num_threads == 1
```

Comment:
is there an performance overhead to starting the Threads -- if so, it's plausible that this might want to be `if num_threads <= RENDER_THREADS_THRESHOLD` <- where that's a small number.

Reply:
Starting even 50 threads is basically free on MRI ruby:

```ruby
require 'benchmark'
puts Benchmark.measure { Array.new(50) { |i| Thread.start { i*i } }.map(&:value) }
#=>  0.002865   0.005927   0.008792 (  0.004879)
```

While it may be different on JRuby, I doubt it's anywhere near the cost of rendering a post.